packages/shared/lib/integrations/scripts/connection/connection.manager.ts

```diff
@@ -0,0 +1,131 @@
+import type { AxiosResponse } from 'axios';
+import type { RecentlyCreatedConnection, Connection, ConnectionConfig } from '../../../models/Connection.js';
+import { LogLevel, LogActionEnum } from '../../../models/Activity.js';
+import { createActivityLogAndLogMessage } from '../../../services/activity/activity.service.js';
+import type { HTTP_VERB } from '../../../models/Generic.js';
+import proxyService from '../../../services/proxy.service.js';
+import connectionService from '../../../services/connection.service.js';
+import environmentService from '../../../services/environment.service.js';
+import metricsManager, { MetricTypes } from '../../../utils/metrics.manager.js';
+
+import * as postConnectionHandlers from './index.js';
+
+interface PostConnectionHandler {
+    (internalNango: InternalNango): Promise<void>;
+}
+
+type PostConnectionHandlersMap = { [key: string]: PostConnectionHandler };
+
+const handlers: PostConnectionHandlersMap = postConnectionHandlers as unknown as PostConnectionHandlersMap;
+
+export interface InternalNango {
+    proxy: ({ method, endpoint, data }: { method?: HTTP_VERB; endpoint: string; data?: unknown }) => Promise<AxiosResponse>;
+    updateConnectionConfig: (config: ConnectionConfig) => Promise<ConnectionConfig>;
+}
+
+async function execute(createdConnection: RecentlyCreatedConnection, provider: string) {
+    const { connection_id, environment_id, provider_config_key } = createdConnection;
+    try {
+        const accountId = await environmentService.getAccountIdFromEnvironment(environment_id);
+        const { success, response: connection } = await connectionService.getConnectionCredentials(
+            accountId as number,
+            environment_id,
+            connection_id,
+            provider_config_key
+        );
+
+        if (!success || !connection) {
+            return;
+        }
+
+        const internalConfig = {
+            environmentId: createdConnection.environment_id,
+            isFlow: true,
+            isDryRun: false,
+            throwErrors: false,
+            connection: connection as unknown as Connection
```

Comment:
I don't think you need the `as unknown as Connection` here and below. `connection` is already of type `Connection` and you already dealt with the null case on L37

Reply:
Good catch, thanks, updated!